client/src/proxy.rs

```diff
@@ -0,0 +1,85 @@
+use base64::prelude::*;
+use hyper::header;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use tokio::net::TcpStream;
+use tonic::transport::Channel;
+use tonic::transport::Endpoint;
+use tower::{service_fn, Service};
+
+/// Options for HTTP CONNECT proxy.
+#[derive(Clone, Debug)]
+pub struct HttpConnectProxyOptions {
+    /// The host:port to proxy through.
+    pub target_addr: String,
+    /// Optional HTTP basic auth for the proxy.
+    pub basic_auth: Option<(String, String)>,
```

Comment:
Definitely ought to say what the two string fields are

Reply:
Fixed